ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -236,46 +238,10 @@ private void setSerdeOptions(final Sink sink) {
       analysis.setSerdeOptions(serdeOptions);
     }
 
-    /**
-     * Get the list of select expressions that are <i>not</i> for meta and key columns in the source
-     * schema.
-     *
-     * <p>Currently, the select expressions can include metadata and key columns, which are later
-     * removed by {@link io.confluent.ksql.planner.plan.KsqlStructuredDataOutputNode}. When building
-     * the {@link SerdeOptions} its important the final schema is used, i.e. the one with these
-     * meta and key columns removed.
-     *
-     * <p>This is weird functionality, but maintained for backwards compatibility for now.
-     *
-     * @return the list of column names in the sink that are not meta or key columns.
-     */
-    private List<ColumnName> getNoneMetaOrKeySelectAliases() {
```

Comment:
The complex logic in here is no longer needed as persistent query schema's can't have unaliased `ROWTIME` or `ROWKEY` in their output schemas.

Reply:
Is this true for joins because we automatically prepend a `<SOURCE>_` to ROWKEY/ROWTIME in the column alias?

Reply:
Yes, this works for JOINs.  Joins have aliased  `ROWKEY` and `ROWTIME`, where as non-joins have unaliased, which is not allowed any more.  This method was previously only filtering out unaliased `ROWKEY` and `ROWTIME` columns. This was necessary because these column actually got removed latter by the `KsqlStructuredDataOutputNode` and so not removing them here would cause a disconnect on the column count.

There are extensive QTT tests that cover join and non-join use of aliased and unaliased `ROWKEY` and `ROWTIME`. Check'em out.